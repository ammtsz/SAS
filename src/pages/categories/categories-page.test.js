import React from "react";
import Categories from "./categories-page.component";

import { shallow } from "enzyme";

describe("<Categories />", () => {
  const getCategories = jest.fn();
  const categories = [
    { id: 1, name: "Sports" },
    { id: 2, name: "Music" },
  ];
  const userDatas = {};
  const resetReport = jest.fn();
  const resetQuiz = jest.fn();

  const mockProps = {
    getCategories,
    categories,
    userDatas,
    resetReport,
    resetQuiz,
  };

  const wrapper = shallow(<Categories {...mockProps} />);

  it("to have two categories cards", () => {
    expect(wrapper.find("[data-testid='categories-cards']")).toHaveLength(2);
  });

});
