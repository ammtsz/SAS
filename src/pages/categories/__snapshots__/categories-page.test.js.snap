// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component <Categories /> shoud render Categories 1`] = `
<DocumentFragment>
  <section
    class="sc-iBPRYJ brIDSy"
    data-testid="categories-page"
    id="test-categories"
  >
    <h1
      class="sc-fubCfw jcLlnp"
      id="test-categories2"
    >
      Categories 
      <small>
        (3)
      </small>
    </h1>
    <section
      class="sc-pFZIQ cQsANz"
    >
      <div
        aria-label="Books start quiz"
        class="sc-bdfBwQ sc-dlfnbm cXyUsK haOKjT"
        data-readonly="true"
        data-testid="category-card"
        role="div"
        tabindex="-1"
      >
        <div
          class="sc-eCssSg caVUlX"
        >
          <div
            aria-label="0 questions answered"
            class="sc-jSgupP eMsPRm progress-bar"
            data-testid="progress-bar"
          >
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
          </div>
          <span
            aria-label="continue Books quiz (0 questions answered)"
            class="category-card__link"
            role="button"
            tabindex="0"
          >
            continue
          </span>
        </div>
        <h2
          class="sc-hKgILt juEGes"
        >
          Books
        </h2>
      </div>
      <div
        aria-label="History start quiz"
        class="sc-bdfBwQ sc-dlfnbm cXyUsK haOKjT"
        data-readonly="true"
        data-testid="category-card"
        role="div"
        tabindex="-1"
      >
        <div
          class="sc-eCssSg caVUlX"
        >
          <div
            aria-label="0 questions answered"
            class="sc-jSgupP eMsPRm progress-bar"
            data-testid="progress-bar"
          >
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
          </div>
          <span
            aria-label="continue History quiz (0 questions answered)"
            class="category-card__link"
            role="button"
            tabindex="0"
          >
            continue
          </span>
        </div>
        <h2
          class="sc-hKgILt juEGes"
        >
          History
        </h2>
      </div>
      <div
        aria-label="Sports start quiz"
        class="sc-bdfBwQ sc-dlfnbm cXyUsK haOKjT"
        data-readonly="true"
        data-testid="category-card"
        role="div"
        tabindex="-1"
      >
        <div
          class="sc-eCssSg caVUlX"
        >
          <div
            aria-label="0 questions answered"
            class="sc-jSgupP eMsPRm progress-bar"
            data-testid="progress-bar"
          >
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
            <span
              class="sc-gKsewC eDsWGz  "
              data-testid=" "
            />
          </div>
          <span
            aria-label="continue Sports quiz (0 questions answered)"
            class="category-card__link"
            role="button"
            tabindex="0"
          >
            continue
          </span>
        </div>
        <h2
          class="sc-hKgILt juEGes"
        >
          Sports
        </h2>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
